ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/AccountTrieNodeDataRequest.java

```diff
@@ -28,23 +29,28 @@
 
 class AccountTrieNodeDataRequest extends TrieNodeDataRequest {
 
-  AccountTrieNodeDataRequest(final Hash hash) {
-    super(RequestType.ACCOUNT_TRIE_NODE, hash);
+  private final boolean isMainAccountTrie;
```

Comment:
Do we need `isMainAccountTrie`?  There is only one set of accounts, and the account storage is wrapped up in the StorageTrieNode requests.

Reply:
So if I understand correctly, to know if we are in the main account trie vs if we are going down an account storage trie we just have to look at the type of the request ?

main account trie -> AccountTrieNodeDataRequest
account storage trie -> StorageTrieNodeDataRequest

And if it's true this information is already present and it is the RequestType field